controllers/machinedeletionremediation_controller.go

```diff
@@ -122,14 +124,20 @@ func (r *MachineDeletionRemediationReconciler) Reconcile(ctx context.Context, re
 				finalErr = utilerrors.NewAggregate([]error{updateErr, finalErr})
 			}
 			finalResult.RequeueAfter = time.Second
+		} else {
+			// if Status.Condition Processing is False, the remediation is completed and visitedMachine map can be
+			// cleared out
+			if meta.IsStatusConditionPresentAndEqual(remediation.Status.Conditions, commonconditions.ProcessingType, metav1.ConditionFalse) {
```

Comment:
IIUC we can use a nicer function -> `IsStatusConditionTrue` instead of `IsStatusConditionPresentAndEqual`

Reply:
I should've looked better, I noticed only now that internally `IsStatusConditionTrue` calls `IsStatusConditionPresentAndTrue` :facepalm: